root/static/scripts/edit/URLCleanup.js

```diff
@@ -1296,12 +1296,14 @@ const CLEANUPS: CleanupEntries = {
   },
   'cdjapan': {
     match: [new RegExp(
-      '^(https?://)?(www\\.)?cdjapan\\.co\\.jp/(product|person)/',
+      '^(https?://)?(www\\.)?cdjapan\\.co\\.jp/(detailview|product|person)',
       'i',
     )],
     restrict: [LINK_TYPES.mailorder],
     clean: function (url) {
-      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?cdjapan\.co\.jp\/(person|product)\/([^\/?#]+)(?:.*)?$/, 'https://www.cdjapan.co.jp/$1/$2');
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?cdjapan\.co\.jp\//, 'https://www.cdjapan.co.jp/');
+      url = url.replace(/^https:\/\/www\.cdjapan\.co\.jp\/detailview\.html\?KEY=([^\/?#]+)(?:.*)?$/, 'https://www.cdjapan.co.jp/product/$1');
+      url = url.replace(/^https:\/\/www\.cdjapan\.co\.jp\/(person|product)\/([^\/?#]+)(?:.*)?$/, 'https://www.cdjapan.co.jp/$1/$2');
```

Comment:
* Since the scheme and the domain name are unchanged, they can be captured to shorten the replacement.
* That was in the old code already but the last non-capturing group is unneeded: `(?:.*)?` is the same as `.*`.
```suggestion
      url = url.replace(/^(https:\/\/www\.cdjapan\.co\.jp)\/detailview\.html\?KEY=([^\/?#]+).*$/, '$1/product/$2');
      url = url.replace(/^(https:\/\/www\.cdjapan\.co\.jp)\/(person|product)\/([^\/?#]+).*$/, '$1/$2/$3');
```

Reply:
Sounds good. I'll remove the other (?:.*)? regexes on a separate commit.